src/extensions.rs

```diff
@@ -91,23 +91,47 @@ impl SystemTimeExt for SystemTime {
 
 pub trait MimeExt {
     fn is_compressed_format(&self) -> bool;
+    fn guess_chartset(&self) -> Option<mime::Name>;
```

Comment:
Typo here, should be `guess_charset` not `guess_chartset`. This will need to be changed for all references too.